inc/CLI/BuildCommand.php

```diff
@@ -23,58 +23,121 @@
 	 *     $ wp traduttore build 123
-	 *     ZIP file generated for translation set (ID: 1)
-	 *     ZIP file generated for translation set (ID: 3)
-	 *     ZIP file generated for translation set (ID: 7)
+	 *     Language pack generated for translation set (ID: 1)
+	 *     Language pack generated for translation set (ID: 3)
+	 *     Language pack generated for translation set (ID: 7)
+	 *     Success: Language pack generation finished
+	 *
+	 *     # Generate language packs for all projects.
+	 *     $ wp traduttore build --all
+	 *     Language pack generated for translation set (ID: 1)
+	 *     Language pack generated for translation set (ID: 2)
+	 *     Language pack generated for translation set (ID: 3)
+	 *     Language pack generated for translation set (ID: 4)
+	 *     Language pack generated for translation set (ID: 5)
+	 *     Language pack generated for translation set (ID: 7)
+	 *     Success: Language pack generation finished
 	 *
 	 * @since 2.0.0
 	 *
 	 * @param array $args Command args.
 	 * @param array $assoc_args Associative args.
 	 */
 	public function __invoke( $args, $assoc_args ) {
-		$locator = new ProjectLocator( $args[0] );
-		$project = $locator->get_project();
+		$all      = get_flag_value( $assoc_args, 'all', false );
+		$force    = get_flag_value( $assoc_args, 'force', false );
+		$projects = $this->check_optional_args_and_all( $args, $all );
 
-		if ( ! $project ) {
-			WP_CLI::error( 'Project not found' );
+		if ( ! $projects ) {
+			return;
 		}
 
-		$translation_sets = (array) GP::$translation_set->by_project_id( $project->get_id() );
+		foreach ( $projects as $project ) {
+			if ( ! $project ) {
+				continue;
+			}
 
-		/* @var GP_Translation_Set $translation_set */
-		foreach ( $translation_sets as $translation_set ) {
-			$zip_provider = new ZipProvider( $translation_set );
+			$translation_sets = (array) GP::$translation_set->by_project_id( $project->get_id() );
 
-			if ( ! get_flag_value( $assoc_args, 'force' ) && $translation_set->last_modified() <= $zip_provider->get_last_build_time() ) {
-				WP_CLI::log( sprintf( 'No ZIP file generated for translation set as there were no changes (ID: %d)', $translation_set->id ) );
+			/* @var GP_Translation_Set $translation_set */
+			foreach ( $translation_sets as $translation_set ) {
+				$zip_provider = new ZipProvider( $translation_set );
 
-				continue;
+				if ( ! $force && $translation_set->last_modified() <= $zip_provider->get_last_build_time() ) {
+					WP_CLI::log( sprintf( 'No language pack generated for translation set as there were no changes (ID: %d)', $translation_set->id ) );
+
+					continue;
+				}
+
+				if ( $zip_provider->generate_zip_file() ) {
+					WP_CLI::log( sprintf( 'Language pack generated for translation set (ID: %d)', $translation_set->id ) );
+
+					continue;
+				}
+
+				WP_CLI::warning( sprintf( 'Error generating Language pack for translation set (ID: %d)', $translation_set->id ) );
```

Comment:
I ran the command I got this warning in a few places. The reason for this was the those languages did not have any translations. Maybe we could add another information if there is no translations.

Reply:
Do you mean no translations, or no strings at all? Because the `ZipProvider` only returns `false` when there are no strings (entries) at all.

We could certainly do something like this:

```php
if ( 0 === $translation_set->current_count() ) {
    WP_CLI::log( sprintf( 'No language pack generated for translation set as it\'s empty (ID: %d)', $translation_set->id ) );

    continue;
}
```

If that's what you mean…?

Reply:
I was getting errors for Spanish, English and Portuguese 
 for https://translate.required.com/projects/prohelvetia/ph-content-types/

Reply:
Okay, yeah than it's that.

Because there are no entries, `$zip_provider->generate_zip_file()` returns `false` here:

https://github.com/wearerequired/traduttore/blob/cc7798e4441de6135d8e68fbba037fdf2921f0c5/inc/ZipProvider.php#L124-L128

That's why you get the warning. I'll add my suggested change to change the message.

Reply:
Yes, the code looks good. I am not sure about the message.